characters_counter/characters_counter.py

```diff
@@ -1,16 +1,59 @@
-from collections import Counter
+import re
+from collections import Counter, defaultdict
 from functools import cache
+from pathlib import Path
 
 REQUIREMENT_OCCURRENCE = 1
+NUM_OF_MOST_COMMON_CHARACTERS = 3
 
 
 @cache
 def count_characters_occurrence(string: str) -> int:
     """
-    The function returns the number of characters in the string occuring only
-    once.
+    The function returns the number of characters in the string occuring only once.
     """
     characters_counts = Counter(string)
     unique_occurrences = filter(lambda occurrence: (occurrence == REQUIREMENT_OCCURRENCE), characters_counts.values())
     unique_characters_number = sum(unique_occurrences)
     return unique_characters_number
+
+
+def get_frequent_characters_per_sentence(text: str) -> defaultdict[str, Counter[str]]:
+    """
+    The helper function counts the frequency of each character in the input sentence, excluding spaces.
+    Single word is considered as a sentence as well.
+    The number of characters returned is controlled by the global constant "NUM_OF_MOST_COMMON_CHARACTERS".
+    """
+    sentences = re.split(r"[.?!]", text)
+    character_frequency: defaultdict = defaultdict(Counter)
```

Comment:
Why defaultdict? I think regular dict will work well

Reply:
So, if it is a regular dict, I need to make a few changes...
At first the idea was like this: the count of characters in identical strings is summed up. That's why I had such intricate logic.